importer/Uploader.py

```diff
@@ -28,17 +28,49 @@ def make_aliases(self):
                 new_aliases.append(new_alias)
         return new_aliases
 
-    def add_labels(self, target_item, labels, log):
+    def add_labels(self, target_item, labels, aliases, log):
+        """
+        Add labels and aliases.
+
+        Normally, if the WD item already has a label in $lang
+        and the data object has another one, the new one
+        will be automatically added as an alias. Otherwise
+        (no existing label), it will be added as a label.
+
+        However, the data model supports an optional [aliases]
+        field as well.
+        Its contents will be added directly as an alias, without checking
+        if it could be a label first.
+        """
         for label in labels:
             target_item.get()
-            name = labels[label]['value']
-            lang = labels[label]['language']
+            label_content = labels[label]['value']
+            language = labels[label]['language']
             self.wdstuff.addLabelOrAlias(
-                lang, name, target_item)
+                language, label_content, target_item)
             if log:
                 t_id = target_item.getID()
-                message = "{} ADDED LABEL {} {}".format(t_id, lang, name)
+                message = "{} ADDED LABEL {} {}".format(
+                    t_id, language, label_content)
                 log.logit(message)
+        if len(aliases) > 0:
```

Comment:
If you can guarantee that aliases is an iterable then you can skip this step.

If you cannot guarantee it I would probably recommend changing the order of the variables to have `labels` and `aliases` as the last two and have both default to `None`. Then start the function with `labels = labels or {}` and same for aliases. That way you don't have to pass empty values. Alternatively leave as is but add `aliases = aliases or {}` (and optionally `labels = labels or {}`) to the start to allow aliases (and optionally labels) to be `None`.

In both cases you can skip the `len` test